frame/system/src/weights.rs

```diff
@@ -74,3 +65,399 @@ impl ExtrinsicsWeight {
+	/// `MAX(max)`
+	pub max: ForDispatchClass<u32>,
+}
+
+impl Default for BlockLength {
+	fn default() -> Self {
+		BlockLength::max_with_normal_ratio(
+			5 * 1024 * 1024,
+			DEFAULT_NORMAL_RATIO,
+		)
+	}
+}
+
+impl BlockLength {
+	/// Create new `BlockLength` with `max` for every class.
+	pub fn max(max: u32) -> Self {
+		Self {
+			max: ForDispatchClass::new(max),
+		}
+	}
+
+	/// Create new `BlockLength` with `max` for `Operational` & `Mandatory`
+	/// and `normal * max` for `Normal`.
+	pub fn max_with_normal_ratio(max: u32, normal: Perbill) -> Self {
+		let mut len = BlockLength::max(max);
+		len.max.normal = normal * max;
+		len
+	}
+}
+
+/// A struct holding value for each `DispatchClass`.
+#[derive(Clone, Eq, PartialEq, Default, RuntimeDebug, Encode, Decode)]
+pub struct ForDispatchClass<T> {
+	/// Value for `Normal` extrinsics.
+	pub normal: T,
+	/// Value for `Operational` extrinsics.
+	pub operational: T,
+	/// Value for `Mandatory` extrinsics.
+	pub mandatory: T,
+}
+
+impl<T: Copy> ForDispatchClass<T> {
+	/// Create new `ForDispatchClass` with the same value for every class.
+	pub fn new(val: T) -> Self {
+		Self {
+			normal: val,
+			operational: val,
+			mandatory: val,
+		}
+	}
+
+	/// Set the value to given class.
+	pub(crate) fn set(&mut self, value: T, class: ExtrinsicDispatchClass) {
+		match class {
+			ExtrinsicDispatchClass::All => {
+				self.normal = value;
+				self.operational = value;
+				self.mandatory = value;
+			},
+			ExtrinsicDispatchClass::One(DispatchClass::Normal) => {
+				self.normal = value;
+			},
+			ExtrinsicDispatchClass::One(DispatchClass::Operational) => {
+				self.operational = value;
+			},
+			ExtrinsicDispatchClass::One(DispatchClass::Mandatory) => {
+				self.mandatory = value;
+			},
+		}
+	}
+
+	/// Get value for given class.
+	pub fn get(&self, class: DispatchClass) -> T {
+		match class {
+			DispatchClass::Normal => self.normal,
+			DispatchClass::Operational => self.operational,
+			DispatchClass::Mandatory => self.mandatory,
+		}
+	}
+}
+#[derive(Default, RuntimeDebug)]
+pub struct ValidationErrors {
+	pub has_errors: bool,
+	#[cfg(feature = "std")]
+	pub errors: Vec<String>,
+}
+
+#[cfg(feature = "std")]
+macro_rules! error_assert {
+	($cond : expr, $err : expr, $format : expr $(, $params: expr )*) => {
+		if !$cond {
+			$err.has_errors = true;
+			$err.errors.push(format!($format $(, &$params )*));
+		}
+	}
+}
+#[cfg(not(feature = "std"))]
+macro_rules! error_assert {
+	($cond : expr, $err : expr, $format : expr $(, $params: expr )*) => {
+		if !$cond {
+			$err.has_errors = true;
+		}
+	}
+}
+
+pub type ValidationResult = Result<BlockWeights, ValidationErrors>;
+
+/// A ratio of `Normal` dispatch class within block, used as default value for
+/// `BlockWeight` and `BlockLength`. The `Default` impls are provided mostly for convenience
+/// to use in tests.
+const DEFAULT_NORMAL_RATIO: Perbill = Perbill::from_percent(75);
+
+/// Block weight limits configuration.
```

Comment:
To not clash with what is `ExtrinsicsWeight ` and `ForDispatchClass`, similar to my previous comment, I would name this somehow to emphasise that it is used for to _configure_ the _limits_, not to be mistaken with something that accumulates weights for example.